core/src/main/java/org/jruby/javasupport/JavaMethod.java

```diff
@@ -196,92 +181,92 @@ public final int hashCode() {
-    }
-
-    @JRubyMethod(rest = true)
-    public IRubyObject invoke(ThreadContext context, IRubyObject[] args) {
-        checkArity(context, args.length - 1);
-
-        final IRubyObject invokee = args[0];
-        final Object[] arguments = convertArguments(args, 1);
-
-        if ( invokee.isNil() ) {
-            return invokeWithExceptionHandling(context, method, null, arguments);
-        }
-
-        final Object javaInvokee;
-        if (!isStatic()) {
-            javaInvokee = JavaUtil.unwrapJavaValue(invokee);
-            if ( javaInvokee == null ) {
-                throw context.runtime.newTypeError("invokee not a java object");
-            }
-
-            if ( ! method.getDeclaringClass().isInstance(javaInvokee) ) {
-                throw context.runtime.newTypeError(
-                    "invokee not instance of method's class" +
-                    " (got" + javaInvokee.getClass().getName() +
-                    " wanted " + method.getDeclaringClass().getName() + ")");
-            }
-
-            //
-            // this test really means, that this is a ruby-defined subclass of a java class
-            //
-            if ( javaInvokee instanceof ReifiedJavaProxy &&
-                // don't bother to check if final method, it won't
-                // be there (not generated, can't be!)
-                ! isFinal ) {
-                JavaProxyClass jpc = ((ReifiedJavaProxy) javaInvokee).___jruby$proxyClass();
-                JavaProxyMethod jpm = jpc.getMethod( method.getName(), parameterTypes );
-                if ( jpm != null && jpm.hasSuperImplementation() ) {
-                    return invokeWithExceptionHandling(context, jpm.getSuperMethod(), javaInvokee, arguments);
-                }
-            }
-        }
-        else {
-            javaInvokee = null;
-        }
-
-        return invokeWithExceptionHandling(context, method, javaInvokee, arguments);
-    }
-
-    @JRubyMethod(rest = true)
-    public IRubyObject invoke_static(ThreadContext context, IRubyObject[] args) {
-        checkArity(context, args.length);
-
-        final Object[] arguments = convertArguments(args, 0);
-        return invokeWithExceptionHandling(context, method, null, arguments);
-    }
-
-    @JRubyMethod
-    @SuppressWarnings("deprecation")
-    public IRubyObject return_type(ThreadContext context) {
-        Class<?> klass = method.getReturnType();
-
-        if (klass.equals(void.class)) {
-            return context.runtime.getNil();
-        }
-        return Java.getProxyClass(context.runtime, klass);
-    }
-
-    @JRubyMethod
-    public IRubyObject type_parameters(ThreadContext context) {
-        return Java.getInstance(context.runtime, method.getTypeParameters());
-    }
+//    @JRubyMethod
+//    @Override
+//    public RubyString name(ThreadContext context) {
+//        return context.runtime.newString(method.getName());
+//    }
+//
+//    @JRubyMethod(name = "public?")
```

Comment:
Do you anticipate we may need this code still?  Is this some form of documentation of how we used to do things?  Otherwise you can delete this.

Reply:
oh man, good catch - thought I already deleted these ...